### ISSUE TITLE
False positive of repeated-keyword 

### ISSUE THREAD

**Issue by keis** [PyCQA/pylint#1843]
### Steps to reproduce
1. Lint the following
```
def fun(data):
    pass


base = {'data': 0}
new = {**base, 'data': 1}

fun(**new)
```
2. See checker fail on `fun(**new)`
3. 😭 

### Current behavior

You get this output
```
E:  8, 0: Got multiple values for keyword argument 'data' in function call (repeated-keyword) 
```

### Expected behavior

This is, in fact, not repeating keyword arguments and should not result in any message

### pylint --version output
pylint 1.8.1, 
astroid 1.6.0
Python 3.6.3 (default, Oct  9 2017, 12:07:10) 
[GCC 7.2.1 20170915 (Red Hat 7.2.1-2)]

**Comment by PCManticore** [comment]
The unpacking in dictionary is a new syntax that this particular check is not aware of, so it makes sense why it emits the error today on this particular code.

**Comment by brycepg** [comment]
I think this is actually an issue with the inference system @PCManticore, astroid thinks the dictionary has items with the same key:

```python
call = astroid.extract_node("""
def fun(data):
    pass


base = {'data': 0}
new = {**base, 'data': 1}

fun(**new) #@
""")
```

> `print(call.repr_tree())`
```
Call(
   func=Name(name='fun'),
   args=[],
   keywords=[Keyword(
         arg=None,
         value=Name(name='new'))])
```
```python
[inferred_dict] = call.keywords[0].value.inferred()
```
> `inferred_dict.repr_tree()`

```Dict(items=[[Const(value='data'), Const(value=0)], [Const(value='data'), Const(value=1)]])"```

**Comment by jab** [comment]
I just hit this with latest pylint.

Here is the code:
```python
DEFAULTS = dict(max_examples=500, deadline=None, timeout=unlimited)
settings.register_profile('default', **DEFAULTS)
settings.register_profile('max_examples_5000', **dict(
    DEFAULTS, max_examples=5000, perform_health_check=False))
```
Here is the spurious error it causes:
```
$ pylint tests/test_hypothesis.py
Using config file .../bidict/.pylintrc
************* Module test_hypothesis
E: 35, 0: Got multiple values for keyword argument 'max_examples' in function call (repeated-keyword)

------------------------------------------------------------------
Your code has been rated at 9.81/10 (previous run: 9.77/10, +0.04)

$ pylint --version
Using config file .../bidict/.pylintrc
pylint 1.8.4,
astroid 1.6.3
Python 3.6.5 (default, Mar 30 2018, 06:42:10)
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]
```

Isn't this expected to be fixed in the latest pylint?

**Comment by brycepg** [comment]
It should be fixed on astroid master. There's been a few standing issues preventing a release of 2.0 astroid

**Comment by jab** [comment]
Ah, I saw that https://github.com/PyCQA/astroid/pull/481 had been merged 6 weeks ago but a new astroid release came out 12 days ago and just assumed that had the fix. Thanks for letting me know that has to wait till 2.0. Just subscribed to https://libraries.io/pypi/astroid so I'll get a notification when 2.0 comes out.

**Comment by jab** [comment]
I just hit this again with astroid 2.0.4, which has the fix for this issue, no?

The code in question:
```python
NOCHECK_SLOW = (HealthCheck.hung_test, HealthCheck.too_slow)
PROFILE_DEFAULT = {
    'max_examples': 200,
    'deadline': None,
    'timeout': unlimited,
    # Enabling coverage slows down hypothesis.
    'suppress_health_check': NOCHECK_SLOW if getenv('COVERAGE') else (),
}
PROFILE_MORE_EXAMPLES = dict(PROFILE_DEFAULT, max_examples=2000, suppress_health_check=NOCHECK_SLOW)
settings.register_profile('DEFAULT', **PROFILE_DEFAULT)
settings.register_profile('MORE_EXAMPLES', **PROFILE_MORE_EXAMPLES)
```

Here is the pylint output:
```
tests/test_hypothesis.py:45:0: E1132: Got multiple values for keyword argument 'suppress_health_check' in function call (repeated-keyword)
tests/test_hypothesis.py:45:0: E1132: Got multiple values for keyword argument 'max_examples' in function call (repeated-keyword)
```

Line 45 refers to the last line of code above. Pylint version 2.1.1.

Reopen?
